### PR TITLE
networkd: Neighbor IPv6 support for LinkLayerAddress

### PR DESCRIPTION
**Pull request by 1848**
Hi,
this is a followup for #13015 (you guys are too fast, was merged before I came home ;) ).

Please double check, I am not at home and could not test but I had a proof of concept working with earlier code.

First commit adds support for ipv6 lladdr.
Second commit is just an alternative approach (if you prefer it).

Thanks

### CODE REVIEW

File: src/network/networkd-neighbor.c [hunk 1]
```diff
@@ -223,6 +219,7 @@ int config_parse_neighbor_lladdr(
         Network *network = userdata;
         _cleanup_(neighbor_free_or_set_invalidp) Neighbor *n = NULL;
         int r;
+        int fam;
```
Comment (yuwata) [inline src/network/networkd-neighbor.c]: Please do not use abbreviated name.
`int family, r;`

File: src/network/networkd-neighbor.h [hunk 2]
```diff
@@ -33,7 +26,7 @@ struct Neighbor {
                 struct ether_addr mac;
                 union in_addr_union ip;
         } lladdr;
-        NeighborLLAddressType lladdr_type;
+        int lladdr_size;
```
Comment (yuwata) [inline src/network/networkd-neighbor.h]: `size_t`

**Comment by yuwata** [comment]
Since my comments are nitpicky. I fixed @1848's commit and squashed the two commit. Also, a simple testcase is added. Setting green label as my change is almost trivial.

@1848 your original commits are saved at https://github.com/yuwata/systemd/commits/neigh_ipv6-original.

**Comment by keszybz** [comment]
```
======================================================================
ERROR: test_neighbor_gre (__main__.NetworkdNetworkTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/test-network/systemd-networkd-tests.py", line 1675, in test_neighbor_gre
    self.wait_online(['dummy98:degraded', 'gretun97:routable', 'ip6gretun97:routable'], timeout='40s')
  File "./test/test-network/systemd-networkd-tests.py", line 346, in wait_online
    check_output(*args, env=env)
  File "./test/test-network/systemd-networkd-tests.py", line 43, in check_output
    return subprocess.check_output(command, universal_newlines=True, **kwargs).rstrip()
  File "/usr/lib64/python3.6/subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "/usr/lib64/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/usr/lib/systemd/systemd-networkd-wait-online', '--timeout=40s', '--interface=dummy98:degraded', '--interface=gretun97:routable', '--interface=ip6gretun97:routable']' returned non-zero exit status 1.
```

**Comment by keszybz** [comment]
Code changes look good though.

**Comment by yuwata** [comment]
@keszybz Thank you. The root of the CI failure is that ip6gre tunnel does not support LinkLocalAddressing= on old kernel. I hope now the CI passes. PTAL.

**Comment by yuwata** [comment]
Finally all CentOS CIs are green. Merging.

**Pull request merged**